packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -122,6 +122,18 @@ export async function fetchHelper(
 					{ driverVersion },
 				);
 			}
+
+			if (!response.ok && response.redirected) {
+				// Return redirected response to allow callers of fetchHelper to handle the redirect.
+				const redirectHeaders = headersToMap(response.headers);
+				return {
+					content: response,
```

Comment:
This does not look right. If it's an error, we should throw here. Other consumers (other than getFileLinkCore) will expect success path to be in a format they expect, not an error response here.
I think we should craft exception with a proper information that is sufficient for redirect flow, but not return success.